File: miniRuby/C++/interpreter/command/IfCommand.cpp
#include "IfCommand.h"

IfCommand::IfCommand(int line, BoolExpr* cond, Command* thenCommands)
    : Command(line), m_cond(cond), m_thenCommands(thenCommands), m_elseCommands(NULL){
    };

IfCommand::~IfCommand(){
  delete m_cond;
  delete m_thenCommands;
  delete m_elseCommands;
}

void IfCommand::setElseCommands(Command* elseCommands){
  m_elseCommands = elseCommands;
};

void IfCommand::Execute(){
  if(m_cond->expr()){
    m_thenCommands->execute();
  }
  else if(m_elseCommands!=NULL){
    m_elseCommands->execute();
  }
}

File: miniRuby/JAVA/interpreter/command/IfCommand.java
package interpreter.command;
import interpreter.expr.BoolExpr;
import interpreter.util.Exit;

public class IfCommand extends Command {
    private BoolExpr cond;
    private Command thenCmds;
    private Command elseCmds;

    public IfCommand(int line, BoolExpr cond, Command thenCmds, Command elseCmds) {
        super(line);
        this.thenCmds = thenCmds;
        this.cond = cond;
        this.elseCmds = elseCmds;
    }

    public void setElseCommands(Command elseCommand) {
        this.elseCmds = elseCommand;
    }

    @Override
    public void execute() {
        if(this.cond.expr()){
            this.thenCmds.execute();
        }else {
            if(this.elseCmds != null) {
                this.elseCmds.execute();
            }
        }
    }
}
